api4/plugin.go

```diff
@@ -161,7 +161,11 @@ func getWebappPlugins(c *Context, w http.ResponseWriter, r *http.Request) {
 	clientManifests := []*model.Manifest{}
 	for _, m := range manifests {
 		if m.HasClient() {
-			clientManifests = append(clientManifests, m.ClientManifest())
+			manifest := m.ClientManifest()
+
+			// There is no reason to expose the SettingsSchema in this API call; it's not used in the webapp.
+			manifest.SettingsSchema = nil
```

Comment:
Since there is no copy as far as I can tell, I think this might be modifying the version stored in the plugin enviroment's `sync.Map`. You might need to make a copy.

Reply:
@crspeller I thought so too, but since `m.ClientManifest()` makes a shallow copy of the manifest, the pointer (which I'm setting to nil) is a copy. That shouldn't change the original pointer's value, correct?  (guilty admission: I never fully learned C, so I'm still non-confident around pointers)

In an earlier version I had put this in a conditional and only set it to nil if the requester was not a sys admin. I hit it multiple times with sys admins and non-sys admins, and it was never erased --- sys admins would always get the settings schema, non-sys admins would not.

Reply:
Ah yes you are right. It's copied there.